pandas/io/parsers.py

```diff
@@ -1176,13 +1176,18 @@ def _should_parse_dates(self, i):
         if isinstance(self.parse_dates, bool):
             return self.parse_dates
         else:
-            name = self.index_names[i]
+            if self.index_names is not None:
+                name = self.index_names[i]
```

Comment:
@jorisvandenbossche this fixes, though not quite sure why this could/would be ``None`` here.

@gfyoung any idea?